packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -21,11 +45,84 @@ class Datalist extends Component {
 	 */
 	onChange(event, payload) {
 		const payloadWithSchema = { ...payload, schema: this.props.schema };
+		this.callTrigger(event);
 		this.props.onChange(event, payloadWithSchema);
 		this.props.onFinish(event, payloadWithSchema);
 	}
 
+	onFocus(event) {
+		this.callTrigger(event);
+	}
+
+	onBlur(event) {
+		this.callTrigger(event);
+	}
+
+	getTitleMap() {
```

Comment:
i think this should be extracted outside of the component instance and split, this way memoization will be easier to achieve